### PR TITLE
[TA] Leverage http pipeline position

### PR DESCRIPTION
**Pull request by mssfang**
- Added clientOptions() to client builder
- Leverage http pipeline position. fixes: https://github.com/Azure/azure-sdk-for-java/issues/15640

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java [hunk 1]
```diff
@@ -69,27 +72,34 @@
...
+    private static final String DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+    private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
     private static final String NAME = "name";
+    private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+    private static final String TEXT_ANALYTICS_PROPERTIES = "azure-ai-textanalytics.properties";
     private static final String VERSION = "version";
-    private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
-    private static final String DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
 
+    private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
```
Comment (alzimmermsft) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java:85]: Does TextAnalytics retun a `retry-after-ms` header? Was looking at their REST API docs and their Swagger definitions and I didn't see anything about `retry-after-ms`.

**Review by alzimmermsft — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java (hunk 1 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java:85]: As I checked. there is no retry-after-ms header returned in the json file. And it doesn't exist in the Swagger definition as well.
Probably having it when I copied the context from AppConfiguration

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java [hunk 2]
```diff
@@ -69,27 +71,27 @@
  */
 @ServiceClientBuilder(serviceClients = {TextAnalyticsAsyncClient.class, TextAnalyticsClient.class})
 public final class TextAnalyticsClientBuilder {
+    private static final String DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+    private static final String NAME = "name";
     private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
-    private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
```
Comment (anuchandy) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java:73]: Hi @johanste Shawn noticed that we don't have a way to tell the Text Analytics cognitive service to echo "x-ms-client-request-id" in the response, i.e., service won't honor "x-ms-return-client-request-id", which means the caller cannot correlate the request with a response.

We want to check with you that this is something we need to communicate to the service team?  Is this something we expect the service to follow, or is it optional?

[Shawn is also checking behavior of Form Recognizer meantime]

**Review by anuchandy — APPROVED** [review]
I didn't mean to block the pr with the question I asked; that's something we can follow up with the service team separately (if needed based on the response from Johan).

**Pull request merged**